d3ploy/tester.py

```diff
@@ -218,44 +218,54 @@ def supply_demand_dict_nond3ploy(sqlite, commod, demand_eq=0):
     return all_dict
 
 
-def residuals(all_dict):
-    """ Conducts a chi2 goodness of fit test
+def residuals_under(all_dict):
+    """ obtains the cumulative under supply
     Parameters
     ----------
     dict_demand: timeseries dictionary of demand values
     dict_supply: timeseries dictionary of supply values
     Returns
     -------
-    returns an int of the chi2 (goodness of fit value) for
-    the two input timeseries dictionaries
+    returns a double as the cumulative difference between demand
+    and suppy, when the demand is larger than supply.
     """
 
     dict_demand = all_dict['dict_demand']
     dict_supply = all_dict['dict_supply']
 
-    start = int(list(dict_demand.keys())[0])
-    demand_total = 0
-    for x in range(start - 1, len(dict_demand)):
-        y = x + 1
+    SSres = 0
+    for x in dict_demand.keys():
         try:
-            demand_total += dict_demand[y]
+            if dict_supply[x] <= dict_demand[x]:
+                SSres += (dict_demand[x] - dict_supply[x])
         except KeyError:
-            demand_total += 0
-    demand_mean = (1 / len(dict_demand)) * demand_total
-    SStot = 0
+            SSres += 0
+    return SSres
+
+
+def residuals_over(all_dict):
+    """ Obtains the cumulative over supply
```

Comment:
Would "cumulative_oversupply" be a more informative name?

In what sense is this cumulative? Cumulative over all time?